main.go

```diff
@@ -58,3 +59,22 @@ func postTasks(c *gin.Context) {
 	tasks = append(tasks, newTask)
 	c.IndentedJSON(http.StatusCreated, newTask)
 }
+
+func putTasks(c *gin.Context) {
+	var newTask task
+	id := c.Param("id")
+
+	if err := c.BindJSON(&newTask); err != nil {
+		return
+	}
+
+	for i := 0; i < len(tasks); i++ {
+		if tasks[i].ID == id {
+			tasks[i] = newTask
+			c.IndentedJSON(http.StatusCreated, newTask)
```

Comment:
Since we're 'updating' and not 'creating', we should serve a statusOk instead of statusCreated